### ISSUE TITLE
Spec disallows `double ~/ double` as constant expression, implementations allow.

### ISSUE THREAD

**Issue by lrhn** [dart-lang/language#3468]
The specification groups `~/` with the shift operators, which is only allowed as constant on `int` operands.

The operator is allowed and well-defined on doubles, and both analyzer and CFE accepts `const c = 3.5 ~/ 2.2;` without complains. 
We should just move it into the same group of constant operations as `-`, `*`, etc., which are defined if the operands are instances of either `int` or `double`.

**Comment by eernstg** [comment]
Good catch, thanks! See https://github.com/dart-lang/language/pull/3495.
